docs/docs/building-a-contact-form.md

```diff
@@ -0,0 +1,140 @@
+---
+title: Building a Contact Form
+---
+
+This guide covers how to create a contact form in a Gatsby site, along with an overview of some strategies for handling form data that has been submitted.
+
+## Adding a Contact Form
+
+Gatsby is built on top of React. So anything that is possible with a React form is possible in Gatsby. Additional details about how to add forms to gatsby can be found in the [Adding Forms](https://www.gatsbyjs.org/docs/adding-forms/) section.
+
+## Creating an Accessible Form
+
+Faulty forms are a common barrier to a website's accessibility, and can be especially problematic if you use a keyboard and screen reader to navigate the web. Forms should be clearly and intuitively organized into groups of related information, and each form field should be identified with a proper label.
+
+More information on creating accessible forms can be found in [WebAIM's article](https://webaim.org/techniques/forms/) on the subject.
+
+## Sending Form Data
+
+When you submit a form, the corresponding data is typically sent to a server to be handled in some manner. More in-depth information on sending form data can be found [on MDN](https://developer.mozilla.org/en-US/docs/Learn/HTML/Forms/Sending_and_retrieving_form_data).
+
+Each method detailed below will start with the following contact form:
+
+```jsx:title=src/pages/contact.js
+<form method="post" action="#">
+  <label>
+    Name
+    <input type="text" name="name" id="name" />
+  </label>
+  <label>
+    Email
+    <input type="email" name="email" id="email" />
+  </label>
+  <label>
+    Subject
+    <input type="text" name="subject" id="subject" />
+  </label>
+  <label>
+    Message
+    <textarea name="message" id="message" rows="5" />
+  </label>
+  <button type="submit">Send</button>
+  <input type="reset" value="Clear" />
+</form>
+```
+
+### Netlify
+
+If you're hosting your site with Netlify, you gain access to their excellent form handling feature. Setting this up only invloves adding a few form attributes:
+
+```jsx:title=src/pages/contact.js
```

Comment:
This could be a good place to use a `diff` file type to show what lines have changed from the example above and new code here.

I also had trouble trying to make this work, I put the default starter on Netlify with this code and I don't actually get submissions to go through: https://amazing-roentgen-d0c1a6.netlify.com/netlify/ I also looked at Netlify's example that worked okay, but it uses plenty of JavaScript even though their docs say you just need to add a few html attributes like this.

Do you have an example where you got this working by chance?

Reply:
I have it working on my site: https://github.com/marcysutton/gatsby-site/blob/master/src/pages/contact.js